tensorflow/compiler/xla/service/conditional_to_select.h

```diff
@@ -0,0 +1,38 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_COMPILER_XLA_SERVICE_CONDITIONAL_TO_SELECT_H_
+#define TENSORFLOW_COMPILER_XLA_SERVICE_CONDITIONAL_TO_SELECT_H_
+
+#include "tensorflow/compiler/xla/service/hlo_module.h"
+#include "tensorflow/compiler/xla/service/hlo_pass_interface.h"
+
+namespace xla {
+
+// A pass which transforms conditionals to selects in places where conditionals
+// are not allowed to appear (e.g. mapped computation)
```

Comment:
Just to be clear, this is a backend constraint right (that we can't codegen this) not a general constraint (i.e. the un-transformed IR still passes the verifier)?  If yes, would be nice to note explicitly.

Reply:
Ok, can you instead s/not allowed/legal but not supported/ or something like that?  I don't want folks to get misled by this comment about what HLO is allowed.